ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import re
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.esi_management import generate_esi, generate_lacp_id, generate_route_target
+from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.network_services.utils import UtilsMixin
+
+
+class PortChannelInterfacesMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def port_channel_interfaces(self) -> dict | None:
+        """
+        Return structured config for port_channel_interfaces
+
+        Only used with L1 network services
+        """
+
+        if not self._network_services_l1:
+            return None
+
+        # Using temp variables to keep the order of interfaces from Jinja
+        port_channel_interfaces = {}
+        subif_parent_interfaces = {}
+
+        for tenant in self._filtered_tenants:
+            if "point_to_point_services" not in tenant:
+                continue
+
+            for point_to_point_service in natural_sort(tenant["point_to_point_services"], "name"):
+                if subifs := point_to_point_service.get("subinterfaces", []):
+                    subifs = [subif for subif in subifs if subif.get("number") is not None]
+                for endpoint in point_to_point_service.get("endpoints", []):
+                    if self._hostname not in endpoint.get("nodes", []):
+                        continue
+
+                    node_index = list(endpoint["nodes"]).index(self._hostname)
```

Comment:
I would probably suggest writing it like this then (if it works also for python 3.8 for the type hint)
```suggestion
                    endpoint_nodes: list = endpoint.get("nodes", [])
                    if self._hostname not in endpoint_nodes:
                        continue

                    node_index = endpoint_nodes.index(self._hostname)
```

Here we are recreating a list for nothing
```
>>> t = [1,2,3]
>>> id(t)
4384968000
>>> tt = list(t)
>>> id(tt)
4384994752
```